app/src/main/java/org/literacyapp/authentication/MergeThread.java

```diff
@@ -123,32 +123,35 @@ private synchronized void findSimilarStudentsUsingMeanFeatureVector(PreProcessor
         // Iterate through all StudentImageCollectionEvents
         List<StudentImageCollectionEvent> studentImageCollectionEvents = studentImageCollectionEventDao.loadAll();
```

Comment:
Can we change to a QueryBuilder only looking for events where the Student is not null?